src/service/invoice-service.ts

```diff
@@ -137,39 +164,48 @@ export default class InvoiceService {
       transfer: invoice.transfer ? TransferService.asTransferResponse(invoice.transfer) : undefined,
       description: invoice.description,
       pdf: invoice.pdf ? invoice.pdf.downloadName : undefined,
-      currentState: InvoiceService.asInvoiceStatusResponse(invoice.latestStatus),
+      currentState: InvoiceService.asInvoiceStatusResponse(invoice.invoiceStatus[invoice.invoiceStatus.length - 1]),
       city: invoice.city,
       country: invoice.country,
       postalCode: invoice.postalCode,
       street: invoice.street,
+      totalInclVat: DineroTransformer.Instance.from(totalInclVat).toObject(),
     };
   }
 
   /**
    * Parses an Invoice Object to a InvoiceResponse
    * @param invoice - The Invoice to parse
    */
-  public static asInvoiceResponse(invoice: Invoice)
-    : InvoiceResponse {
+  public async asInvoiceResponse(invoice: Invoice)
+    : Promise<InvoiceResponse> {
+    const customEntries = invoice.invoiceEntries.map(InvoiceService.asInvoiceEntryResponse);
+    const relations: FindOptionsRelations<SubTransactionRow> = {
+      product: {
+        vat: true,
+      },
+    };
+    const subTransactionRows = await this.manager.find(SubTransactionRow, { where: { invoice: { id: invoice.id } }, relations });
```

Comment:
Why don't you just create an inverse relation from an `Invoice` to its `SubTransactionRows`? That saves this (in my opinion) pretty ugly query in a response builder.